api/app/controllers/spree/api/address_book_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  module Api
+    class AddressBookController < Spree::Api::BaseController
+      # Note: the AddressBook is the resource to think about here, not individual addresses
+
+      before_filter :ensure_api_token
+
+      def mine
+         @user_addresses = current_api_user.user_addresses
+      end
+
+      def update
+        address_params = address_book_params
+        default_flag = address_params.delete(:default)
+        address = current_api_user.save_in_address_book(address_params, default_flag)
+        if address.valid?
+          @user_addresses = current_api_user.user_addresses
+          render :mine, status: :ok
```

Comment:
Ok, so first of all Law of Demeter. Your code shoud be simple, do only one thing. In that case responsibility of rendering resource is in other action, so redirect user to correct resource to get it.

Second case could sounds strange, but I know from experience, that some servers has garbage collector enabled between requests. So faster response (with redirection) means faster garbage collection and freeing server resources.

And third - If I want use API without getting resources rendered on every update, I can disable follow redirects, because after all updates I'll pull it in one request. It's also edge case.